update users in ES after changing group ownership

snopoke opened the pull request:
https://dimagi-dev.atlassian.net/browse/SAASP-10262

snopoke committed:
update users in ES after changing group ownership

dannyroberts commented on corehq/apps/groups/views.py:
```diff
@@ -168,14 +167,21 @@ def _update_group_membership(request, domain, group_id):
     group.save()
+
+    for user in safe_users:
+        update_user_in_es(None, user)
```
We must have been doing this elsewhere as well, right? Is it clear what's broken with the current way of getting group changes into users' es docs? If not, I'm worried that that system may overwrite these changes if it's asynchronous and wrong, or generally that we'll have two systems to debug instead of one.

snopoke replied:
The other place this should be happening is in the `group-pillow`: https://github.com/dimagi/commcare-hq/blob/f73f823081e15bb755f85321ffa104f2530e4243/corehq/pillows/groups_to_user.py#L100

I have not investigated to see if that's not working or perhaps just behind.

dannyroberts replied:
@snopoke I think I figured it out in the commit I added to this PR. The group pillow make use of `group.removed_users` to figure out which users need to be updated, but this view was circumventing the code that adds to or removes from `removed_users`.

In the same PR I removed the line that eagerly puts them into elasticsearch. I wasn't totally sure about this, but I decided to do that so there's one clear way the information is supposed to make it to the User docs in ES.

dannyroberts reviewed (approved):
Thanks for going out of your way to look into a bug a SaaS user reported. Had a question about the solution but overall seems like a reasonable approach

dannyroberts committed:
Properly use Group.removed_users to get updates to ES

This also removes the eager propagation to ES, with the motivation that
while correct, it adds a second way for user's group info to get updated
in ES and adds another layer to reasoning about it.

dannyroberts commented:
Thanks @snopoke for the first pass. Also the fact that this view was previously pulling all user_ids in the project is insane.

snopoke commented:
I can't approve my own PR so :+1: from me

dannyroberts commented:
Thanks!
